pkg/internal/token/serviceprincipaltokencertificate.go

```diff
@@ -90,6 +90,7 @@ func (p *servicePrincipalToken) getPoPTokenWithClientCert(
 		p.cloud.ActiveDirectoryAuthorityHost,
 		p.clientID,
 		p.tenantID,
```

Comment:
what is the use case of this method getPoPTokenWithClientCert? This is not being used by CSP, however wondering whether we need to expose InstanceDiscovery  option here as well..